### PR TITLE
#48 add frame paddings

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: savant/deepstream/buffer_processor.py [hunk 1]
```diff
@@ -123,29 +119,35 @@ def prepare_input(self, buffer: Gst.Buffer):
                         model_uid,
                         class_id,
                     ) = self._model_object_registry.get_model_object_ids(obj_key)
+                    xc = obj_meta['bbox']['xc']
+                    yc = obj_meta['bbox']['yc']
+                    if self._frame_params.padding:
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply savant/deepstream/buffer_processor.py]: Placed shifting after scaling

File: savant/deepstream/metadata.py [hunk 2]
```diff
@@ -31,15 +30,27 @@ def nvds_obj_meta_output_converter(
         if nvds_obj_meta.tracker_confidence < 1.0:  # specified confidence
             confidence = nvds_obj_meta.tracker_confidence
 
+    if frame_params.padding and not frame_params.padding.keep:
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply savant/deepstream/metadata.py]: Removed negation out of conditions

File: savant/deepstream/metadata.py [hunk 3]
```diff
@@ -31,15 +30,27 @@ def nvds_obj_meta_output_converter(
...
+        frame_width = frame_params.width
+        frame_height = frame_params.height
+    else:
+        frame_width = frame_params.total_width
+        frame_height = frame_params.total_height
+
     # scale bbox to [0..1]
     if rect_params.width == 0:
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply savant/deepstream/metadata.py:42]: Added TODO

File: savant/deepstream/metadata.py [hunk 4]
```diff
@@ -31,15 +30,27 @@ def nvds_obj_meta_output_converter(
...
         rbbox = nvds_get_rbbox(nvds_obj_meta)
+        x_center = rbbox.x_center
+        y_center = rbbox.y_center
+        if frame_params.padding and not frame_params.padding.keep:
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply savant/deepstream/metadata.py]: Placed shifting after scaling

**Review by denisvmedyantsev — APPROVED** [review]

**Pull request merged**